File: frontend/src/types/aggregation.ts
export interface CountItem {
  name: string;
  count: number;
}

export interface TrendItem {
  week: string;
  count: number;
}

export interface OverviewData {
  total_reports: number;
  analyzed_count: number;
  sentiment_distribution: Record<string, number>;
  top_industries: CountItem[];
  top_factors: CountItem[];
  report_trend: TrendItem[];
  recent_reports: {
    id: number;
    title: string;
    source?: string | null;
    publish_date?: string | null;
    status: string;
  }[];
}

export interface IndustryAnalysisData {
  industry: string;
  total_reports: number;
  analyzed_count: number;
  sentiment_distribution: Record<string, number>;
  top_factors: CountItem[];
  related_stocks: CountItem[];
  reports: {
    id: number;
    title: string;
    source?: string | null;
    publish_date?: string | null;
    status: string;
  }[];
}

export interface StockAnalysisData {
  stock: string;
  total_reports: number;
  analyzed_count: number;
  sentiment_distribution: Record<string, number>;
  target_prices: string[];
  reports: {
    id: number;
    title: string;
    source?: string | null;
    publish_date?: string | null;
    status: string;
  }[];
}

File: backend/app/schemas/aggregation.py
from pydantic import BaseModel, Field

from app.schemas.report import ReportResponse


class CountItem(BaseModel):
    name: str
    count: int


class TrendItem(BaseModel):
    week: str
    count: int


class OverviewResponse(BaseModel):
    total_reports: int
    analyzed_count: int
    sentiment_distribution: dict[str, int] = Field(default_factory=dict)
    top_industries: list[CountItem] = Field(default_factory=list)
    top_factors: list[CountItem] = Field(default_factory=list)
    report_trend: list[TrendItem] = Field(default_factory=list)
    recent_reports: list[ReportResponse] = Field(default_factory=list)


class IndustryAnalysisResponse(BaseModel):
    industry: str
    total_reports: int
    analyzed_count: int
    sentiment_distribution: dict[str, int] = Field(default_factory=dict)
    top_factors: list[CountItem] = Field(default_factory=list)
    related_stocks: list[CountItem] = Field(default_factory=list)
    reports: list[ReportResponse] = Field(default_factory=list)


class StockAnalysisResponse(BaseModel):
    stock: str
    total_reports: int
    analyzed_count: int
    sentiment_distribution: dict[str, int] = Field(default_factory=dict)
    target_prices: list[str] = Field(default_factory=list)
    reports: list[ReportResponse] = Field(default_factory=list)
